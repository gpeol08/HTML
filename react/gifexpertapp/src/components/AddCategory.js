import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({setCategorias}) => {
	const [inputValue, setInputValue] = useState('');

	const handleChange = (e) => {
		setInputValue(e.target.value);
	};
	const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length>2){
            setCategorias(value => [inputValue,...value]);
            setInputValue('');
        }
	};
	return (
		<form onSubmit={handleSubmit}>
			<input type="text" onChange={handleChange} value={inputValue} />
		</form>
	);
};


AddCategory.propTypes={
    setCategorias:PropTypes.func.isRequired
}