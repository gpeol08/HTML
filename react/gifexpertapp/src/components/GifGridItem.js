import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
	return (
		<div className='card 	'>
			<img src={url} alt={title} />
			<label className="titulo animate__delay-3s">{title}</label>
		</div>
	);
};

GifGridItem.propTypes={
	title: PropTypes.string.isRequired,
	url:PropTypes.string.isRequired
}


