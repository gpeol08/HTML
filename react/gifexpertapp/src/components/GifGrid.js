import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categoria, index }) => {

	const {data,loading} =useFetchGifs(categoria);

	return (
		<>
			<h4 key={categoria} className='animate__delay-3s'> {categoria} </h4>
			{loading && 'Cargando...'}
			<div className="card-grid">
				{data.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</>
	);
};
