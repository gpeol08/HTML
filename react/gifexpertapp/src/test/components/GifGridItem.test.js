import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
	const title = 'Un titulo';
	const url = 'https://media2.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=c97b409cztnul8gq4ltfq3v1g0h5u8uwxd2pax259sjmsd9s&rid=giphy.gif';
    
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('debe mostrar el componente correctamente', () => {
		expect(wrapper).toMatchSnapshot();
    });
    
    test('div con el titulo',()=>{
        const label = wrapper.find('label');
        expect(label.text().trim()).toBe(title);
    });

    test('debe de obtener la imagen igual al url',()=>{
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
});
