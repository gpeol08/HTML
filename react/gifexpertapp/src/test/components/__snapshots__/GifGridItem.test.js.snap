// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card 	"
>
  <img
    alt="Un titulo"
    src="https://media2.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?cid=c97b409cztnul8gq4ltfq3v1g0h5u8uwxd2pax259sjmsd9s&rid=giphy.gif"
  />
  <label
    className="titulo animate__delay-3s"
  >
    Un titulo
  </label>
</div>
`;
